src/address-manager/README.md

```diff
@@ -0,0 +1,49 @@
+
+## Announce Addresses
+
+In some scenarios, a libp2p node will need to announce addresses that it is not listening on. In other words, Announce Addresses are an amendment to the Listen Addresses that aim to enable other nodes to achieve connectivity to this node.
+
+Scenarios for Announce Addresses include:
+- when you setup a libp2p node in your private network at home, but you need to announce your public IP Address to the outside world;
+- when you want to announce a DNS address, which maps to your public IP Address.
+
+## No Announce Addresses
+
+While we need to add Announce Addresses to enable peers' connectivity, we can also not announce addresses that will not be reachable. This way, No Announce Addresses should be specified so that they are not announced by the peer as addresses that other peers can use to dial it.
+
+As stated into the Listen Addresses section, Listen Addresses might get modified after libp2p transports get in action and use them to listen for new connections. This way, libp2p should also take into account these changes so that they can be matched when No Announce Addresses are being filtered out for advertising addresses.
+
+## Implementation
+
+When a libp2p node is created, the Address Manager will be populated from the provided addresses through the libp2p configuration. Once the node is started, the Transport Manager component will gather the listen addresses from the Address Manager, so that the libp2p transports use them to listen on.
+
+Libp2p will use the the Address Manager as the source of truth when advertising the peers addresses. After all transports are ready, other libp2p components/subsystems will kickoff, namely the Identify Service and the DHT. Both of them will announce the node addresses to the other peers in the network. The announce and noAnnounce addresses will have an important role here and will be gathered by libp2p to compute its current addresses to advertise everytime it is needed.
+
+## Future Considerations
+
+### Dynamic address modifications 
+
+In a future iteration, we can enable these addresses to be modified in runtime. For this, the Address Manager should be responsible for notifying interested subsystems of these changes, through an Event Emitter.
+
+#### Modify Listen Addresses
+
+While adding new addresses to listen on runtime is a feasible operation, removing one listen address might have bad implications for the node, since all the connections using that listen address will be closed. With this in mind and taking also into consideration the lack of good use cases for removing listen addresses, the Address Manager API only allows libp2p users to add new Listen Addresses on runtime.
```

Comment:
A future use case for removing listening addresses would be Relay and Rendezvous. If we asked used Rendezvous to find relays to listen on, we could add them to our list. If the relay we found was at its maximum listener capacity, it could potentially tell us to `go away`, at which point we would likely remove it from our listen addresses.

I don't think we need to speculate on the operations here though. Mentioning the potential of them changing in runtime and that it should provide a mechanism for listening to those changes would be sufficient here.